packages/amplify-category-auth/provider-utils/awscloudformation/triggers/VerifyAuthChallengeResponse/captcha-verify.js

```diff
@@ -2,23 +2,16 @@
 const axios = require('axios');
 /* eslint-enable */
 
-exports.handler = (event, context, callback) => {
-  axios
-    .post(
-      `https://www.google.com/recaptcha/api/siteverify?secret=${process.env.RECAPTCHASECRET}&response=${event.request.challengeAnswer}`,
-      {}
-    )
-    .then(response => {
-      if (response && response.data && response.data.success) {
-        event.response.answerCorrect = true;
-        callback(null, event);
-      } else {
-        event.response.answerCorrect = false;
-        callback(new Error('captcha verification error'), event);
-      }
-    })
-    .catch(() => {
-      event.response.answerCorrect = false;
-      callback(new Error('captcha verification error'), event);
-    });
+exports.handler = async (event, context, callback) => {
+  const response = await axios.post(
+    `https://www.google.com/recaptcha/api/siteverify?secret=${process.env.RECAPTCHASECRET}&response=${event.request.challengeAnswer}`,
+    {},
+  );
+
+  if (!(response && response.data && response.data.success)) {
+    event.response.answerCorrect = false;
+    throw new Error('captcha verification error');
+  }
+
```

Comment:
So in an async handler, we are good to just throw any errors that come up. Before, the logic would resolve with `callback()` if the response was in the correct form. Here, we flip the logic - if the response is *not* correct, throw, otherwise proceed as usual.

I should have documented this, that is my fault, I will also make what is happening here clearer.

Reply:
See:

https://github.com/aws-amplify/amplify-cli/blob/63e647bb257b8bee4a5bdc12d6902b1375a4da12/packages/amplify-category-auth/provider-utils/awscloudformation/triggers/VerifyAuthChallengeResponse/captcha-verify.js#L10-L21